src/components/FeatureToggleProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useMemo } from 'react';
+
+type Props = { children?: React.ReactNode };
+type FeatureToggles = Array<string>;
+
+const FEATURE_FLAGS_STORAGE = 'node_featureFlags';
+
+export const FeatureToggleContext = createContext<FeatureToggles>([]);
+
+export const FeatureToggleProvider: React.FC<Props> = ({ children }) => {
+  const featureFlags = useMemo(() => {
+    try {
+      const storageItem = localStorage.getItem(FEATURE_FLAGS_STORAGE) || '[]';
```

Comment:
Curious to know, wouldn't this allow users to modify the flag locally and test out the feature that might not be production ready and behind a flag which could have unexpected behaviour?

Reply:
Sure, but they would need to know the feature flag. And yet, even if they knew and set, that isn't really a problem as it would only change the experience for them.

This is just the Node.js website, not a critical application that would be negatively impacted by those feature flags 👀

Reply:
What do you think about using an env variable or maybe something like feature flag object which works exactly like how it is implemented now. 🤔 

I know it's not a critical problem but still user being able to alter the data and see what they aren't supposed to see is also what I find not appropriate. 

Anyways, as I am not a reviewer here just trying to get engaged in what's happening around. I hope you don't mind. 😅

Reply:
I wouldn't really go to such lengths. Not see what they're not supposed to see? This is not a company lol.

Feature flags should not be env based as we want to allow selected people to be able to test sole features. Like the i18n feature switch. It is literally 0 deal, 0 problem if other people see it.

Because anyways, this is an open source repository. The source is here. So there's no point in trying to hide things :)

Reply:
Yeah I agree, it's open sourced and all transparent but my point was not about hiding but more about not allowing user to enable certain flag when it was not supposed to, that's why we have the flags isn't it. It's just my thoughts.

Reply:
> about not allowing user to enable certain flag when it was not supposed to

I don't see any problem with this. What exactly do you think will happen as a "bad" thing? Maximum, they're able to see features in development. That's it. There's no harm done, and the only reason we've created this, to begin with, is not for security reasonings. It's just so that we don't hinder our users with non-finished development features.

It is also important to mention that by using an `env` variable, you're either enabling it to the whole audience or no one. We don't want to have specific extra environments with specific features enabled. For that, we just have the Preview Branches.

We're also not going to make A/B testing or crowd-control; this is just a simple mechanism to allow random people (or not so random) to preview certain things and contribute to our development.

For example, whilst a specific language is not released, e.g., Portuguese, we don't want to show Portuguese to everybody, so we could use a feature flag here. Just to simply hide it.

Reply:
I understand they're just your thoughts, but as you said, you're jumping here out of the blue and making assumptions without knowing the reasoning behind this PR. It is of course my fault for not sharing exactly the why's of this PR, so I hope my comments here are detailed enough 🙈